spec/models/feed_event_spec.rb

```diff
@@ -63,4 +63,129 @@
       end
     end
   end
+
+  describe "after_save .update_article_counters_and_scores" do
+    let!(:article) { create(:article) }
+    let!(:user1) { create(:user) }
+    let!(:user2) { create(:user) }
+
+    it "updates article counters and scores when a FeedEvent is saved" do
+      # Create some feed events to simulate behavior
+      create(:feed_event, category: "impression", article: article, user: user1)
+      create(:feed_event, category: "click", article: article, user: user1)
+      create(:feed_event, category: "reaction", article: article, user: user1)
+      create(:feed_event, category: "comment", article: article, user: user2)
+
+      # Trigger the after_save
+      create(:feed_event, category: "impression", article: article, user: user2)
+
+      # Reload the article to get the updated counters and scores
+      article.reload
+
+      expect(article.feed_success_score).to eq((1 + 5 + 10) / 2.0) # Calculated score
```

Comment:
nit: May be worth to referring to the variable itself here, ie `FeedEvent::REACTION_SCORE_MULTIPLIER + FeedEvent::COMMENT_SCORE_MULTIPLIER +...` that way the test won't go stale easily.

Reply:
Good call